### ISSUE TITLE
`deprecated()` doesn't have a fallback defined if the version is not passed

### ISSUE THREAD

**Issue by cjfields** [bioperl/bioperl-live#263]
`Bio::Root::RootI::deprecated()` doesn't indicate what the expected behavior should be if the version isn't defined.  This was brought up indirectly due to failures using `prove -w` as in #257.

```
[cjfields@Chriss-MacBook-Air bioperl-live (master)]$ prove -w t/Root/RootI.t
t/Root/RootI.t .. 1/61
#   Failed test 'simple'
#   at t/Root/RootI.t line 96.
# found Bioperl warning: Use of uninitialized value $throw_version in substitution (s///) at Bio/Root/RootI.pm line 233.
# found Bioperl warning:
# --------------------- WARNING ---------------------
# MSG: Test1
# STACK Bio::Root::RootI::deprecated Bio/Root/RootI.pm:254
# STACK Test::Warn::warnings_like t/Root/RootI.t:96
# STACK toplevel t/Root/RootI.t:96
# ---------------------------------------------------
# expected to find warning: (?^:Test1)

#   Failed test at t/Root/RootI.t line 97.
# found Bioperl warning: Use of uninitialized value $throw_version in substitution (s///) at Bio/Root/RootI.pm line 233.
# found Bioperl warning:
# --------------------- WARNING ---------------------
# MSG: Test2
# STACK Bio::Root::RootI::deprecated Bio/Root/RootI.pm:254
# STACK Test::Warn::warnings_like t/Root/RootI.t:97
# STACK toplevel t/Root/RootI.t:97
# ---------------------------------------------------
# expected to find warning: (?^:Test2)

#   Failed test 'simple'
#   at t/Root/RootI.t line 115.
# found Bioperl warning: Use of uninitialized value $throw_version in substitution (s///) at Bio/Root/RootI.pm line 233.
# found Bioperl warning:
# --------------------- WARNING ---------------------
# MSG: Test1
# STACK Bio::Root::RootI::deprecated Bio/Root/RootI.pm:254
# STACK Test::Warn::warnings_like t/Root/RootI.t:115
# STACK toplevel t/Root/RootI.t:115
# ---------------------------------------------------
# expected to find warning: (?^:Test1)

#   Failed test 'simple'
#   at t/Root/RootI.t line 116.
# found Bioperl warning: Use of uninitialized value $throw_version in substitution (s///) at Bio/Root/RootI.pm line 233.
# found Bioperl warning:
# --------------------- WARNING ---------------------
# MSG: Test2
# STACK Bio::Root::RootI::deprecated Bio/Root/RootI.pm:254
# STACK Test::Warn::warnings_like t/Root/RootI.t:116
# STACK toplevel t/Root/RootI.t:116
# ---------------------------------------------------
# expected to find warning: (?^:Test2)
# Looks like you failed 4 tests of 61.
t/Root/RootI.t .. Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/61 subtests

Test Summary Report
-------------------
t/Root/RootI.t (Wstat: 1024 Tests: 61 Failed: 4)
  Failed tests:  15-16, 22-23
  Non-zero exit status: 4
Files=1, Tests=61,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.20 cusr  0.04 csys =  0.28 CPU)
Result: FAIL
```

Note these fail only with `-w` set, due to the `$throw_version` variable being undefined.  

This method was originally designed to warn about upcoming potential incompatible changes, which requires a version, so using something like:

```
Bio::Root::Root->deprecated('Foo')
```

where a version isn't passed is not explicitly defined.  My thought is this should be silently converted to a `throw()`, maybe creating a default message if one isn't passed in.

**Comment by cjfields** [comment]
Implemented on a branch (https://github.com/bioperl/bioperl-live/tree/issues/263-deprecated).  If there aren't any objections after a few weeks I'll merge.

**Comment by hlapp** [comment]
I think this issue is moot now with the recent changes by @carandraug?

**Comment by carandraug** [comment]
With `deprecated` being deprecated this indeed quite pointless but. But I pushed the fix anyway in 12f27ee
